applications/openshift/ocp_idp_no_htpasswd/rule.yml

```
@@ -14,3 +14,13 @@ severity: medium
 warnings:
     - general: |-
         {{{ openshift_cluster_setting("/apis/config.openshift.io/v1/oauths/cluster") | indent(8) }}}
+
+template:
+    name: yamlfile_value
+    vars:
+        file: cluster.yaml
+        yamlpath: ".spec.identityProviders[:].type"
+        value: "HTPasswd"
+        entity_check: "at least one"
+        negate: True
```

the build system doesn't support using directly `True` here when building with `python2`. We need to use something like:
```suggestion
        negate: 'yes'
```

---

I remember having some sort of documentation talking about this, but I can't find it. Maybe @matejak

---

@ggbecker  I also can't find it.
I think it's caused by this: https://github.com/ComplianceAsCode/content/blob/141f810e048bb0dfbc15ae162c76ba6ee5f39078/ssg/yaml.py#L27
This can be relevant: https://github.com/ComplianceAsCode/content/pull/2737

---

Yes, back at the day when we moved to YAML, there was pushback against the implicit conversion of bool strings to actual bools, so it is turned off. The YAML parsing lib IMO goes too far - it converts both keys and values. But at the same time, turning this off for our content may violate expectations as well as cause problems like you have experienced. The template documentation should be clear that one should use quoted strings as arguments.

---

Done and done.